attribute/set_test.go

```diff
@@ -225,3 +226,32 @@ func args(m reflect.Method) []reflect.Value {
 	}
 	return out
 }
+
+var result attribute.Set
+
+func BenchmarkNewSet(b *testing.B) {
+	attrs := []attribute.KeyValue{
+		attribute.Bool("bool true", true),
+		attribute.Bool("bool false", false),
+		attribute.Int64("int64 1", 1),
+		attribute.Int64("int64 -1", -1),
+		attribute.Float64("float64 10", 10),
+		attribute.Float64("float64 -10", -10),
+		attribute.String("string empty", ""),
+		attribute.String("hello", "world"),
+		attribute.BoolSlice("[]bool", []bool{true, false, true}),
+		attribute.Int64Slice("[]int64", []int64{-10, 13124, -23, 0, 2}),
+		attribute.Float64Slice("[]float64", []float64{10.23, 941.1, 184e9, -2.3}),
+		attribute.StringSlice("[]string", []string{"", "one", "two"}),
+	}
+	// Pre-sort to remove from first iteration results.
```

Comment:
Doesn't this defeat the purpose of the benchmark? Aren't we interested in finding out the performance of a typical case where the attributes are not necessarily sorted? Or is being sorted more typical?